lib/matplotlib/axes/_axes.py

```diff
@@ -940,12 +948,29 @@ def axline(self, xy1, xy2, **kwargs):
         axhline : for horizontal lines
         axvline : for vertical lines
         """
+        def _to_points(xy1, xy2, slope):
+            """
+            Check for a valid combination of input parameters and convert
+            to two points, if necessary.
+            """
+            if (xy2 is None and slope is None or
+                    xy2 is not None and slope is not None):
+                raise TypeError(
+                    "Exactly one of 'xy2' and 'slope' must be given")
+            if xy2 is None:
+                x1, y1 = xy1
+                xy2 = (x1, y1 + 1) if np.isinf(slope) else (x1 + 1, y1 + slope)
+            return xy1, xy2
 
         if "transform" in kwargs:
             raise TypeError("'transform' is not allowed as a kwarg; "
                             "axline generates its own transform")
-        x1, y1 = xy1
-        x2, y2 = xy2
+        if slope is not None and (self.get_xscale() != 'linear' or
+                                  self.get_yscale() != 'linear'):
```

Comment:
You _could_ change the scale after calling this and it would "work" without error. Catching this case is difficult in this implementation since `slope` is thrown away, so I think we should at least call it "undefined behaviour" in the docs.